qualtran/cirq_interop/t_complexity_protocol.py

```diff
@@ -94,6 +94,11 @@ def _is_clifford_or_t(stc: Any, fail_quietly: bool) -> Optional[TComplexity]:
     if not isinstance(stc, (cirq.Gate, cirq.Operation)):
         return None
 
+    if isinstance(stc, cirq.GlobalPhaseGate) or (
+        isinstance(stc, cirq.Operation) and isinstance(stc.gate, cirq.GlobalPhaseGate)
+    ):
+        return TComplexity()
```

Comment:
I think ideally this shouldn't be required but the T-complexity protocol is being updated in PR https://github.com/quantumlib/Qualtran/pull/825 so this hack can be removed there. 

We should update the `CIRQ_TYPE_TO_BLOQ_MAP` in `_cirq_gate_to_bloq` to return `GlobalPhase` bloq when it sees a `cirq.GlobalPhaseGate`. That would be in-line with other substitutions that we do for basic gates.

Reply:
`_cirq_gate_to_bloq` already converts `GlobalPhase` (merged with https://github.com/quantumlib/Qualtran/pull/828)
https://github.com/quantumlib/Qualtran/blob/4dffc97de898b92509934c593e6ffb9b77b9337d/qualtran/cirq_interop/_cirq_to_bloq.py#L347-L348

The only neccessary hack here is `isinstance(stc, cirq.Operation) and isinstance(stc.gate, cirq.GlobalPhaseGate)` because cirq decompose seems to convert it to a `global_phase_operation` instead of a gate. I updated the code now